simpleviewinc/keg-hub
# Clean up stories

## Context
* I want to write component stories using MDX
* I want to add documentation notes to the components
* Currently we can only show the components, and not add documentation within story book

## Goal
* Update storybook to allow writing stories in MDX


## Updates
* `repos/keg-components/.storybook/main.js`
  * Updated the config to allow pull in MDX files
  * Cleaned up presets
* `repos/keg-components/.storybook/preview.js`
  * Added code to pretty print the JSDoc notes for each component
  * This is not complete, but will be added to when I have time
  * For now it does nothing
* `repos/keg-components/configs/webpack.config.js`
  * Removed un-used loader for `.jsx` files
* `repos/keg-components/package.json`
  * Updated storybook patch version
* `repos/keg-components/src/components/**.stories.js`
  * Updated to use `/` instead of `|` for the `storiesOf` functions
  * Removes a warning about it from the inspector console
* `repos/keg-components/src/components/touchable/touchable.js`
  * Added propTypes to the component
* `repos/keg-components/src/components/touchable/touchable.stories.mdx`
  * Added touchable stories in MDX
  * Gives an example of using the `className` prop

## Testing
* **TODO - Add a docker package url. My docker is having issues right now. I'll add it once I get it working again.**
* Pull the branch `keg && keg pr `
* Run keg-components `keg components start --build --cache false`
* Go to https://local.kegdev.xyz
* Click on the `touchable` stories
  * Ensure it renders properly
  * Check the inspector, and ensure there's no Storybook separator error
